lms/djangoapps/grades/tasks.py

```diff
@@ -138,7 +146,7 @@ def _has_database_updated_with_new_score(
         if api_score is None:
             # Same case as the initial 'if' above, for submissions-specific scores
             return score_deleted
-        reported_modified_time = api_score.created_at
+        reported_modified_time = api_score['created_at']
```

Comment:
The only test we'd had before was `test_retry_subsection_grade_on_no_sub_score`, which mocked the whole return value as `None` and so didn't hit the happy-path case of having a dictionary return value.